src/main.rs

```diff
@@ -5,10 +5,6 @@ use vector::app::Application;
 
 #[cfg(unix)]
 fn main() {
-    // Configure our tracking allocator.
-    #[cfg(feature = "allocation-tracing")]
-    vector::internal_telemetry::allocations::init_allocation_tracing();
```

Comment:
Seems like we still need to actually call this? Otherwise the reporter thread never gets spawned, and so on. We could move it to `src/app.rs`, though, where you've added the new code for enabling tracing if the CLI flag is passed.

Reply:
Oh yeah did miss that. I think this should go stay in main.